src/chainlit/frontend/src/api/index.tsx

```diff
@@ -0,0 +1,67 @@
+import toast from 'react-hot-toast';
+
+const devServer = 'http://127.0.0.1:8000';
+const url = import.meta.env.DEV ? devServer : window.origin;
+const serverUrl = new URL(url);
+
+const httpEndpoint = `${serverUrl.protocol}//${serverUrl.host}`;
+export const wsEndpoint = httpEndpoint;
+
+const api = {
+  fetch: async (
+    method: string,
+    endpoint: string,
+    token?: string,
+    data?: any,
+    signal?: AbortSignal
+  ) => {
+    try {
+      const headers: { Authorization?: string; 'Content-Type': string } = {
+        'Content-Type': 'application/json'
+      };
+      if (token) headers['Authorization'] = `Bearer ${token}`;
+
+      const res = await fetch(httpEndpoint + endpoint, {
+        method,
+        headers,
+        signal,
+        body: JSON.stringify(data)
+      });
+
+      if (!res.ok) {
+        const body = await res.json();
+
+        toast.error(res.statusText);
+        throw new Error(
```

Comment:
the thrown error will be caught by the catch statement that is just beneath. Then it will only print the status text and not the full error with the details right?

That's why the previous error handling was looking like this:

```tsx
 if (!res.ok) {
      let err: ClientError;
      try {
        const body = await res.json();
        err = new ClientError(res.statusText, body.detail);
      } catch (_) {
        err = new ClientError(res.statusText);
      }
      throw err;
}
```

Reply:
The previous error handling was only for `const body = await res.json();` because there was no try/catch at all. Now, it's already wrapper by a try/catch, so if `res.json()` isn't it normal to have it catch by the global one ?